torque-core/src/main/kotlin/com/workday/torque/TestRunFactory.kt

```diff
@@ -61,34 +59,18 @@ class TestRunFactory {
                                 "${testSession.failedCount} failed, took " +
                                 "${testSession.durationMillis.millisToHoursMinutesSeconds()}."
                     )
+                    pullDeviceFiles(args, filePuller)
 
                     testSession
                 }
         ).asSingle(Dispatchers.Default)
     }
 
-    private fun CoroutineScope.pullChunkTestFiles(
-            args: Args, filePuller: FilePuller,
-            deviceTestsResults: List<AdbDeviceTestResult>
-    ) {
+    private fun CoroutineScope.pullDeviceFiles(args: Args, filePuller: FilePuller) {
         if (args.testFilesPullDeviceDirectory.isEmpty() || args.testFilesPullHostDirectory.isEmpty()) {
             return
         }
-        val completedTestResults = deviceTestsResults.filter { it.status !is AdbDeviceTestResult.Status.Ignored }
-        if (completedTestResults.isEmpty()) {
-            return
-        }
 
-        launch {
-            val pullFileTimeout = Timeout(args.installTimeoutSeconds, TimeUnit.SECONDS)
-            completedTestResults
-                    .map { TestDetails(it.className, it.testName) }
-                    .forEach { testDetails: TestDetails ->
-                        filePuller.pullTestFolder(args.testFilesPullDeviceDirectory,
-                                                  args.testFilesPullHostDirectory,
-                                                  testDetails,
-                                                  pullFileTimeout).await()
-                    }
-        }
+        launch { filePuller.pullFolder(args).await() }
```

Comment:
why do you need to launch here?

Reply:
actually might not need to, a relic of the old test-specific calls.